File: app/src/main/java/com/trien/dnflavor/Utils/GifLoader.java
package com.trien.dnflavor.Utils;

import android.content.AsyncTaskLoader;
import android.content.Context;
import android.content.SharedPreferences;

import com.trien.dnflavor.Model.GifImage;

import java.util.List;

/**
 * Loads a list of sizzles by using an AsyncTask to perform the
 * network request to the given URL.
 */

public class GifLoader extends AsyncTaskLoader<List<GifImage>> {

    /** Tag for log messages */
    private static final String LOG_TAG = GifLoader.class.getName();

    private SharedPreferences.OnSharedPreferenceChangeListener sharedPreferenceChangeListener;

    private List<GifImage> mGifImages;
    /**
     * Constructs a new {@link GifLoader}.
     *
     * @param context of the activity
     */
    public GifLoader(Context context) {
        super(context);
    }

    @Override
    protected void onStartLoading() {
        forceLoad();
    }

    /**
     * This is on a background thread.
     */
    @Override
    public List<GifImage> loadInBackground() {

        // Perform the network request, parse the response, and extract a list of gif images.
        List<GifImage> gifImages = GiphyHelper.getGif(getContext());

        return gifImages;
    }
}

File: app/src/main/java/com/trien/dnflavor/Utils/GifLoader.kt
package com.trien.dnflavor.Utils

import android.content.AsyncTaskLoader
import android.content.Context
import android.content.SharedPreferences

import com.trien.dnflavor.Model.GifImage

/**
 * Loads a list of sizzles by using an AsyncTask to perform the
 * network request to the given URL.
 */

class GifLoader
/**
 * Constructs a new [GifLoader].
 *
 * @param context of the activity
 */
(context: Context) : AsyncTaskLoader<List<GifImage>>(context) {

    private val sharedPreferenceChangeListener: SharedPreferences.OnSharedPreferenceChangeListener? = null

    private val mGifImages: List<GifImage>? = null

    override fun onStartLoading() {
        forceLoad()
    }

    /**
     * This is on a background thread.
     */
    override fun loadInBackground(): List<GifImage> {

        // Perform the network request, parse the response, and extract a list of gif images.

        return GiphyHelper.getGif(context)
    }

    companion object {

        /** Tag for log messages  */
        private val LOG_TAG = GifLoader::class.java.name
    }
}
